library/Requests.php

```diff
@@ -826,17 +827,30 @@ public static function decompress($data) {
 			return $data;
 		}
 
-		if (function_exists('gzdecode') && ($decoded = @gzdecode($data)) !== false) {
-			return $decoded;
+		if (function_exists('gzdecode')) {
+			$decoded = @gzdecode($data);
+			if ($decoded !== false) {
```

Comment:
This code not exactly the same. In the original code, the `&&` operator will cause the result of `$decoded = @gzdecode($data)` to be cast as a `bool` before being compared to `false`.

In the changed code, this casting doesn't happen, and considering we do a strict comparison, it would for example have a different result when `$decoded = @gzdecode($data)` would return `null`.

Now, according to the docs, `gzdecode()` would only ever return `false` on failure. But I'm unsure what it would return if you successfully decode a document of zero length. Would it return `null` then?

Reply:
I think the new version is actually more correct, but it would be a BC.

Reply:
> Now, according to the docs, gzdecode() would only ever return false on failure. But I'm unsure what it would return if you successfully decode a document of zero length. Would it return null then?

Basing myself also on the documentation, I would expect it to return an empty string in that case as the standard return type is a string.

Reply:
> This code not exactly the same. In the original code, the && operator will cause the result of $decoded = @gzdecode($data) to be cast as a bool before being compared to false.

Not sure if that would ever make a difference, so I'm not convinced this is a BC-break.
To proof my point: https://3v4l.org/OAoLN